### PR TITLE
[Compose] - 2272 - Add link attachment description line limit

### LINKED ISSUE #2272: [Compose] - Add limit to max lines in a link preview
### Context

Sometimes link previews get super long, typical for YouTube links for example. In UI components, we have a `streamUiLinkDescriptionMaxLines` attribute controlling the max lines displayed in a link preview, its default value is 5.

Example of too lengthy link previews:

image

### Change Proposal

Implement something similar in Compose, and make it configurable if possible.

### Acceptance criteria

To accept this task we need to complete the following items:

- [] Compose SDK limits max lines in link previews
- [] Max lines is configurable
- [] Docs describe this configuration option

### PR DESCRIPTION
**Pull request by filbabic**
Fixes #2272 

### 🎯 Goal

We needed to add a default limit to the number of lines to link Attachments, because some links are too long.

We also needed to add a way to customize this through attachment factories, so that our users can easily change this limit.

### 🛠 Implementation details

Added a parameter to the LinkAttachmentFactory and LinkAttachmentContent to customize this limit. Changed the ˙defaultFactories()˙ to be a function, so that we can pass in a parameter and customize this if needed.

### 🎨 UI Changes

| Before | After |
| --- | --- |
| before| after|

### 🧪 Testing

Send a link that has a lengthy description - e.g. a YouTube link. It should appear with a maximum of 5 lines of description.

### ☑️ Checklist

- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
~~- [ ] New code is covered by unit tests~~
- [x] Comparison screenshots added for visual changes
- [x] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### 🎉 GIF
limit

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/ChatTheme.kt [hunk 1]
```diff
@@ -46,7 +46,7 @@ private val LocalDateFormatter = compositionLocalOf<DateFormatter> {
  * @param colors - The set of colors we provide, wrapped in [StreamColors].
  * @param typography - The set of typography styles we provide, wrapped in [StreamTypography].
  * @param shapes - The set of shapes we provide, wrapped in [StreamShapes].
- * @param attachmentFactories - Attachment factories that we provide. By default, images and files.
```
Comment (zsmb13) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/ChatTheme.kt:49]: Good call 😃

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/LinkAttachmentFactory.kt [hunk 2]
```diff
@@ -44,15 +50,20 @@ public class LinkAttachmentFactory : AttachmentFactory(
...
  * and the onLongItemClick handler.
+ *
+ * @param descriptionLineLimit - The limit of how many lines we show for the link description.
  * */
 @OptIn(ExperimentalFoundationApi::class)
 @Composable
-public fun LinkAttachmentContent(attachmentState: AttachmentState) {
+public fun LinkAttachmentContent(
+    attachmentState: AttachmentState,
+    descriptionLineLimit: Int,
```
Comment (zsmb13) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/LinkAttachmentFactory.kt:60]: What about calling this something that has `maxLines` in it? Would be a familiar name for Android devs, both `TextView` and now `Text` use that as the name for this value.

File: UNRELEASED_CHANGELOG.md [hunk 3]
```diff
@@ -78,10 +78,13 @@
 - Improved the Message list scrolling behavior and scroll to bottom actions
 - Added an unread count on the Message list's scroll to bottom CTA
 - Improved the way we build items in the Message list
+- Added line limit to link attachment descriptions
+- Added a way to customize the default line limit for link descriptions
 
 ### ✅ Added
 
 ### ⚠️ Changed
+- `StreamAttachment.defaultFactories()` is a function now, instead of a property.
```
[the remark this answers is not in the document]
Comment (filbabic) [reply UNRELEASED_CHANGELOG.md:87]: Yup, makes sense!

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/StreamAttachmentFactories.kt [hunk 4]
```diff
@@ -6,13 +6,21 @@ import io.getstream.chat.android.compose.state.messages.attachments.AttachmentSt
 
 public object StreamAttachmentFactories {
 
+    /**
+     * The default description length of link attachments. We limit this, because for some links the description
+     * can prove to be too long.
+     * */
```
[the remark this answers is not in the document]
Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/StreamAttachmentFactories.kt:12]: So one thing that's annoying is that when adding comments this is generated by default. Any way we can change the default format to automate this?

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**